server/src/core/user/user-service.ts

```diff
@@ -104,6 +119,22 @@ export class Users implements UserService {
     validators.validatesNominateBeneficiary(args);
     const { phone, nominator } = args;
     try {
+      const nominatorUser = await this.collection.findOne({ _id: nominator });
+      if (!nominatorUser) throw createResourceNotFoundError(messages.ERROR_USER_NOT_FOUND);
```

Comment:
What error do you suggest? Since in this case, the user doesn't exist in the database?

Reply:
Oh, I was thinking maybe something similar to what you did on line 178 where you used createMiddlemanNominationFailedError... Just wanted to make sure you did this on purpose

Reply:
Aah okay. In this case I opted for the "resource not found" error because this happens before we even try to add the middleman. In practice this error should never occur because the user should correspond to the currently authenticated user, and therefore should have to exist unless we have serious bug in our code or the db data is corrupted.